pipeline/metrics_holder.go

```diff
@@ -18,14 +18,16 @@ type metricsHolder struct {
 	registry           *prometheus.Registry
 }
 
+type counter [2]*prometheus.CounterVec
```

Comment:
Let's have a struct here with `count` and `size` fields instead of slice. It's confusing to choose between indices.